cmd/test-otel-init-go/main_test.go

```diff
@@ -0,0 +1,291 @@
+package main
+
+// testing test-otel-init-go tests otel-init-go by using otel-cli
+// to receive spans and validate things are working
+//
+// this is still very much a work in progress idea and might not fully
+// pan out but the first part is looking good
+//
+// TODOs:
+// [ ] replace that time.Sleep with proper synchronization
+// [ ] use random ports for listener address?
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+// CliEvent is mostly the same as otel-cli's internal event format, with
+// the addition that it has a place to stuff events.
+type CliEvent struct {
+	TraceID    string            `json:"trace_id"`
+	SpanID     string            `json:"span_id"`
+	ParentID   string            `json:"parent_span_id"`
+	Library    string            `json:"library"`
+	Name       string            `json:"name"`
+	Kind       string            `json:"kind"`
+	Start      string            `json:"start"`
+	End        string            `json:"end"`
+	ElapsedMS  int               `json:"elapsed_ms"`
+	Attributes map[string]string `json:"attributes"`
+	Events     []CliEvent        // reader code will stuff kind=event in here
+}
```

Comment:
when https://github.com/equinix-labs/otel-cli/pull/51 merges some of this can go away